lib/src/value/calculation.dart

```diff
@@ -158,6 +382,246 @@ class SassCalculation extends Value {
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value pow(Object base, Object? exponent) {
+    var args =
+        List<Object>.unmodifiable([base, if (exponent != null) exponent]);
+    _verifyLength(args, 2);
+    base = _simplify(base);
+    exponent = exponent.andThen(_simplify);
+    if (base is! SassNumber ||
+        exponent is! SassNumber ||
+        exponent.hasUnit('%') ||
+        base.hasUnit('%')) {
+      return SassCalculation._("pow", args);
+    }
+    _verifyCompatibleNumbers([base, exponent]);
+    return number_lib.pow(base, exponent);
+  }
+
+  /// Creates a `log()` calculation with the given [arguments].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value log(Object number, Object? base) {
+    number = _simplify(number);
+    base = base.andThen(_simplify);
+    var args = [number, if (base != null) base];
+    if (number is! SassNumber ||
+        (base != null && base is! SassString && base is! SassNumber) ||
+        number.hasUnit('%') ||
+        (base is SassNumber && base.hasUnit('%'))) {
+      return SassCalculation._("log", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return number_lib.log(number, base is SassNumber ? base : null);
+  }
+
+  /// Creates a `atan2()` calculation with the given [arguments].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value atan2(Object y, Object? x) {
+    _verifyLength([y, if (x != null) x], 2);
+    y = _simplify(y);
+    x = x.andThen(_simplify);
+    var args = [y, if (x != null) x];
+    if (y is! SassNumber ||
+        x is! SassNumber ||
+        y.hasUnit('%') ||
+        x.hasUnit('%')) {
+      return SassCalculation._("atan2", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return number_lib.atan2(y, x);
+  }
+
+  /// Creates a `rem()` calculation with the given [arguments].
```

Comment:
Here `arguments` should also be replaced, but I wonder if the variables `y` and `x` should be renamed to `dividend` and `divisor` respectively 🤔

Reply:
Great suggestions! Thanks a lot :)